server/auth.go

```diff
@@ -647,48 +647,47 @@ func checkClientTLSCertSubject(c *client, fn func(string) bool) bool {
 	switch {
 	case hasEmailAddresses:
 		for _, u := range cert.EmailAddresses {
-			if fn(u) {
-				c.Debugf("Using email found in cert for auth [%q]", u)
+			if match, ok := fn(u, nil); ok {
+				c.Debugf("Using email found in cert for auth [%q]", match)
 				return true
 			}
 		}
 		fallthrough
 	case hasSANs:
 		for _, u := range cert.DNSNames {
-			if fn(u) {
-				c.Debugf("Using SAN found in cert for auth [%q]", u)
+			if match, ok := fn(u, nil); ok {
+				c.Debugf("Using SAN found in cert for auth [%q]", match)
 				return true
 			}
 		}
 	case hasURIs:
 		for _, u := range cert.URIs {
-			if fn(u.String()) {
-				c.Debugf("Using URI found in cert for auth [%q]", u)
+			if match, ok := fn(u.String(), nil); ok {
+				c.Debugf("Using URI found in cert for auth [%q]", match)
 				return true
 			}
 		}
 	}
 
-	// Try to get the full RDN Sequence that includes the domain components.
-	var rdns pkix.RDNSequence
-	if _, err := asn1.Unmarshal(cert.RawSubject, &rdns); err == nil {
```

Comment:
I will add this back and consider removal in another release

Reply:
But then is that not going to "disable" the new code below (calling ldap.FromCertSubject)?

Reply:
Or will that be a fall through and if auth with that failed, we will still invoke the correct RDN sequence check

Reply:
What I went for is first try the match that has the correct order and the domain component (DC) elements, if that fails then fall through and match with the appended domain components (so not correct order) and if that fails then match the string representation without the domain components.  Eventually I think we will deprecate both last two checks only use the one with the right order to stop relying on ToRDNSequence from `pkix.Name`  /cc @philpennock

Reply:
Might be worth logging at some verbosity level when we do this fall-through, or if that's going to be too spammy then incrementing some varz counters.